### PR TITLE
DB-11809 Implement Feature Store benchmark

### PR DESCRIPTION
**Pull request by OlegMazurov**

### CODE REVIEW

**Review by ascend1 — CHANGES_REQUESTED** [review]
The name of this benchmark is too general. Is it possible to make it more specific? Or if we plan to make it a whole suite for feature store later, maybe we should document it in source at least regarding the following points:

- which queries are benchmarked (and potentially why)
- benchmark setup
- what numbers we are interested in

**Review by hatyo — APPROVED** [review]

**Comment by OlegMazurov** [comment]
@ascend1 This micro-benchmark was developed according to the spec provided outside GitHub/JIRA. Indeed, there are plans to extend it to become a general Feature Store Benchmark.
I have added comments addressing your points above.

File: splice_machine/src/test/java/com/splicemachine/benchmark/FeatureStoreBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+     *    In a table with 10 columns, select columns c2, c4, c6, c8, c10, with 100 columns: c20, c40, c60, c80, c100 and with 1000: c200, c400, c600, c800, c1000.
+     *
+     *    SELECT <tableName>.<columnName>, ...
+     *    FROM <tableName>, ...
+     *    WHERE (<tableName>.<primaryKey>, ...) = (<random value>, ...)
+     *
+     *    Prepared queries are run from multiple concurrent connections for the specified number of operations per connection.
+     *    Query results are read and hashed for possible validation (not implemented).
+     *    Query execution times are collected to compute statistics provided by the general framework.
+     */
```
Comment (martinrupp) [inline splice_machine/src/test/java/com/splicemachine/benchmark/FeatureStoreBenchmark.java:289]: good information. we might want to have this at the top of the file?

Comment (OlegMazurov) [reply splice_machine/src/test/java/com/splicemachine/benchmark/FeatureStoreBenchmark.java:289]: Although it's part of the benchmark spec, provided in a separate document from which this description was borrowed, this comment really explains what _runQueries()_ does. It's similar in that regard to the _createTables()_ comment above, which explains how tables are created. I thought separating comments from what they explain would decrease clarity.

**Review by ascend1 — APPROVED** [review]
Awesome, these comments make it much easier to understand what's going on in this benchmark. Thanks!

**Pull request merged**